modules/schema/src/main/java/org/apache/ignite/internal/schema/registry/SchemaRegistryImpl.java

```diff
@@ -87,28 +83,37 @@ public SchemaRegistryImpl(int initialVer, Function<Integer, SchemaDescriptor> hi
             throw new SchemaRegistryException("Failed to find schema: ver=" + ver);
     }
 
-    /**
-     * Gets schema descriptor for the latest version if initialized.
-     *
-     * @return Schema descriptor if initialized, {@code null} otherwise.
-     * @throws SchemaRegistryException If failed.
-     */
+    /** {@inheritDoc} */
     @Override public @Nullable SchemaDescriptor schema() {
         final int lastVer0 = lastVer;
 
         if (lastVer0 == INITIAL_SCHEMA_VERSION)
```

Comment:
Why do we need this check? If there are some race here - we can return any SchemaDesctiptor (null or first).

Reply:
Without this check the method call may end up with an exception.
However, in LiveSchema mode we have to detect somehow if the schema is non-initialized, to register the first version on the first operation.